### PR TITLE
[AMD] Support ds3.2 on gfx942 platform

### PR DESCRIPTION
**Pull request by wufann**
## Motivation
Support ds3.2 on AMD gfx942 GPUs.

## Modifications
1. Install Tilelang with gfx942-rocm700 arch in rocm.dockerfile.
2. Support gfx942 fp8 e4m3fnuz data type in tilelang act_quant kernel.
3. Reduce LDS usage on gfx942 arch in tilelang sparse_attention_fwd_kernel_v1 kernel.
4. Fix aiter fp8_mqa_logits kernel output name.
5. Modify the tuple input dtype(when enable fused_rms_fp8_group_quant) according to the change in the dtype of indexer weights_proj(PR 17205). 

## Accuracy Tests

```
python3 -m sglang.launch_server --model /models/DeepSeek-V3.2-Exp \
        --tp 8 \
        --port 10086 \
        --trust-remote-code \
        --nsa-prefill-backend tilelang \
        --nsa-decode-backend tilelang

cd benchmark/gsm8k
python3 bench_sglang.py --num-questions 1319 --port 10086 --num-shots 20

# MI300X
Accuracy: 0.956
Invalid: 0.000
Latency: 384.742 s
Output throughput: 334.895 token/s

# MI308X
Accuracy: 0.956
Invalid: 0.000
Latency: 532.694 s
Output throughput: 245.700 token/s

# MI355X
Accuracy: 0.957
Invalid: 0.000
Latency: 280.959 s
Output throughput: 463.010 token/s
```

```
docker build --build-arg SGL_BRANCH=v0.5.7 --build-arg GPU_ARCH=gfx942-rocm700 -t v0.5.7.post2-rocm700-mi30x-test -f rocm.Dockerfile .

[+] Building 6083.9s (26/26) FINISHED 
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
6. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
7. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
8. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/layers/attention/nsa/nsa_indexer.py [hunk 1]
```diff
@@ -8,13 +8,15 @@
...
 from sglang.srt.utils import add_prefix, ceil_align, is_cuda, is_hip, is_npu
 
 global _use_multi_stream
 _is_cuda = is_cuda()
 _is_hip = is_hip()
 _is_npu = is_npu()
+_is_fp8_fnuz = is_fp8_fnuz()
 if _is_cuda:
     try:
         import deep_gemm
```
Comment (hubertlu-tw) [inline python/sglang/srt/layers/attention/nsa/nsa_indexer.py:22]: FYI: There is a breakage introduced by a merged PR recently which may require a fix from this PR:
https://github.com/sgl-project/sglang/pull/17432

**Review by hubertlu-tw — APPROVED** [review]
LGTM

**Comment by yctseng0211** [comment]
> @yctseng0211 shall we also add test/registered/amd/test_deepseek_v32_basic.py for MI325/MI300? If yes, please provide the instructions. If not, we can add it later in a separate PR.

Hi @hubertlu-tw I think we can add it later in a separate PR (8-gpu 325).

File: python/sglang/srt/layers/attention/nsa/nsa_indexer.py (hunk 1 above)
Comment (hubertlu-tw) [reply python/sglang/srt/layers/attention/nsa/nsa_indexer.py:22]: The line number I tagged is wrong btw.

**Review by HaiShaw — APPROVED** [review]
Please simplify docker file in upcoming PR
